src/utiltime.cpp

```diff
@@ -9,22 +9,42 @@
 
 #include "utiltime.h"
 
+#include <chrono>
+
 #include <boost/date_time/posix_time/posix_time.hpp>
 #include <boost/thread.hpp>
 
 using namespace std;
 
 static int64_t nMockTime = 0; //!< For unit testing
+static int64_t nNtpTimeOffset = 0;
 
 int64_t GetTime()
 {
     if (nMockTime) return nMockTime;
 
+    time_t now = time(NULL);
+    assert(now > 0);
+    return now - GetNtpTimeOffset();
+}
+
+int64_t GetTimeNow()
+{
+    if (nMockTime) return nMockTime;
+
     time_t now = time(NULL);
     assert(now > 0);
     return now;
 }
 
+int64_t GetSteadyTime()
```

Comment:
what exactly is steadyTime?

http://en.cppreference.com/w/cpp/chrono/steady_clock

Reply:
as the link describes, is a clock that is not affected by time zone changes, cannot be regressed and every tick corresponds to a physical tick. in some systems is used to measure time since boot. this is the context I'm using it:
- on every cycle of a time interval t, 2 time measures are taken, using the steady and the normal clock each of them.
- on every cycle >0 of a time interval t, the elapsed time is calculated by substracting the value of the previous sample.
- if the elapsed time of the clocks differs between them in a value higher than a predefined threshold (1000ms in the current implementation) it's safe to assume the local clock has been artificially changed so we need to resync against ntp servers before minting a block.

Reply:
cool, thanks for the explanation :)